packages/compass-aggregations/src/components/saved-pipelines/saved-pipelines.tsx

```diff
@@ -110,6 +112,17 @@ export const SavedPipelines = ({
     [editor_view_type, onDeletePipeline, setDeletePipelineId]
   );
 
+  const onOpenPipelineClick = useCallback(
+    (id: string) => {
+      if (shouldConfirmBeforeOpen) {
+        setOpenPipelineId(id);
+      } else {
+        onOpenConfirm(id);
+      }
```

Comment:
I might be missing something but if whether or not this confirm should be shown is always based on redux state maybe it makes sense to move this logic to the action creator itself? 🤔 Like if you'd need to use this action creator in some other place, you probably wouldn't want to also need to always attach the `isModified` state and two dispatchers to make it work, what do you think?

Reply:
that's a good point.
but this change would require us to move the component state to the redux as well and also the tracking logic (which is the reason i actually added this code here).

Reply:
as we are already tracking from redux, I'll move it there :)

Reply:
added in [494bcab](https://github.com/mongodb-js/compass/pull/4068/commits/494bcab554c3ea186c9675a03bc9a0483a9fc4ee)